### PR TITLE
Add UTC timestamps in log lines in Pipelinesteps.echo

### PR DESCRIPTION
**Pull request by georgfedermann**
log statements using other logging mechanisms than Pipelinesteps.echo() will not have a timestamp prefix.

### CODE REVIEW

File: vars/phaseDeploy.groovy [hunk 1]
```diff
@@ -3,21 +3,23 @@ import org.ods.service.OpenShiftService
 import org.ods.service.ServiceRegistry
 import org.ods.util.GitUtil
 import org.ods.util.MROPipelineUtil
+import org.ods.util.PipelineSteps
 import org.ods.util.Project
 
 def call(Project project, List<Set<Map>> repos) {
+    def steps = new PipelineSteps(this)
```
Comment (metmajer) [inline vars/phaseDeploy.groovy]: @georgfederman why not use the service registry here? Would make testing easier, as right now, PipelineSteps is not exchangable with our do-nothing implementation in test/util/PipelineSteps.groovy.

@martsec FYI

**Review by metmajer — CHANGES_REQUESTED** [review]

File: vars/phaseDeploy.groovy (hunk 1 above)
Comment (georgfedermann) [reply vars/phaseDeploy.groovy]: good thought! will do that.

**Comment by metmajer** [comment]
Thanks @georgfedermann! Is this patch exhaustive across all files that use an `echo` (except for `mroPipeline.groovy`) ?

**Comment by georgfedermann** [comment]
> Thanks @georgfedermann! Is this patch exhaustive across all files that use an `echo` (except for `mroPipeline.groovy`) ?

yes, exhaustive over all echo call within ods-mro-jenkins-shared-library, except for echo calls within steps.sh of course.

**Review by martsec — DISMISSED** [review]

**Comment by michaelsauter** [comment]
There's a plugin that does that - https://plugins.jenkins.io/timestamper/. This also allows to see e.g. elapsed time.

Maybe better to rely on that?

**Comment by metmajer** [comment]
@michaelsauter seems like this plugin requires a global configuration. We need to assure that timestamp format is the same across all pipelines. Can this be done using this plugin?

Also, we need this right now to see where the pipeline is running slower than expected (at least get an idea). We can swap this out in a future version.

**Comment by michaelsauter** [comment]
> We need to assure that timestamp format is the same across all pipelines.

Why? Is this a requirement? If so, could we add the reasoning in this PR, or as a comment so that it is clear why this was done? From an outsider's perspective (mine) it looks like you just weren't aware of the plugin ...

To debug performance issues, I highly recommend installing the plugin though. You have full control over the Jenkins instance, so you are free to install any plugin you want. Plus, the plugin gives you elapsed time, which is way more helpful than UTC timestamps ...

FYI @oalyman I believe you installed this plugin a while back in one of your projects and thought about adding it to ODS by default, right?

**Comment by metmajer** [comment]
@michaelsauter I wasn't aware of this plugin and I'd be happy to dig into it if we would have more time (and our 2.x branches in ods-core weren't already frozen).

If we can guarantee a common time base including time zone neutrality (UTC) to be able to compare timings across pipelines and clusters to see where time is spent using this plugin, I am happy to swap out later.

**Review by metmajer — APPROVED** [review]

**Pull request merged**